AtReconstruction/AtPatternRecognition/AtTrackFinderTC.cxx

```diff
@@ -160,11 +178,197 @@ AtPATTERN::AtTrackFinderTC::clustersToTrack(PointCloud &cloud, const std::vector
    for (const auto &point : points)
       retEvent->AddNoise(event.GetHit(point.GetID()));
 
+   for (auto &track : tracks) {
+      if (track.GetHitArray().size() > 0)
+         SetTrackInitialParameters(track);
+      //retEvent->AddTrack(std::move(track));
+   }
+
+   bool kMergeTracks = false;
+
+   std::vector<AtTrack> mergedTracks;
+   if(tracks.size() > 2){
+
+      //const double tolerance = 2.0;
+      std::vector<bool> processed(tracks.size(), false);
+
+      for(Int_t numtr = 0; numtr < tracks.size(); numtr++){
+         std::cout << "Theta " << tracks.at(numtr).GetGeoTheta() << std::endl;
+         if (processed[numtr]) continue;
+       
+         auto track0 = tracks.at(numtr);  
+         auto hitArray0 = track0.GetHitArrayObject();
+         auto theta0 = track0.GetGeoTheta();
+
+         AtTrack newTrack;
+
+         for(auto &hit : hitArray0){
+            newTrack.AddHit(hit);
+                           
+         }
+                     
+         if (numtr + 1 < tracks.size()) {
+                           
+            for (size_t tr = numtr + 1; tr < tracks.size(); ++tr) {
+               if (processed[tr]) continue;
+
+               auto track = tracks.at(tr);
+               auto hitArray = track.GetHitArrayObject();
+               auto theta = track.GetGeoTheta();
+
+               if (std::abs((theta0 * TMath::RadToDeg()) - (theta * TMath::RadToDeg())) <= tolerance) {
+                  for(auto &hit : hitArray){
+                     newTrack.AddHit(hit);
+                                 
+                  }
+                  processed[tr] = true;
+               }
+
+            }  
+         }
+                        
+         processed[numtr] = true;
+         fTrackTransformer->ClusterizeSmooth3D(newTrack, fClusterRadius, fClusterDistance);             
+         mergedTracks.push_back(newTrack);
+
+      }
+
+      kMergeTracks = true;  
+   }
+
+   if(kMergeTracks){
+    std::swap(tracks, mergedTracks);
+   }
+
```

Comment:
Track merging should be controlled by a run-time flag. Changes to core classes like this should retain the old behavior by default (or it could break code for others)